monkey/tests/unit_tests/infection_monkey/puppet/test_puppet.py

```diff
@@ -47,6 +53,9 @@ def test_puppet_run_multiple_payloads():
 
 
 def test_fingerprint_exception_handling(monkeypatch):
-    p = Puppet(agent_event_queue=MagicMock(spec=IAgentEventQueue), plugin_registry=PluginRegistry())
+    p = Puppet(
+        agent_event_queue=MagicMock(spec=IAgentEventQueue),
+        plugin_registry=PluginRegistry(MagicMock()),
+    )
```

Comment:
We should have a unit test where we verify an `UnknownPluginError` is raised if the Island returns a 404.